File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/model/dto/reservation/WalkInOptionResponse.java
package com.hcmute.reservation.model.dto.reservation;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class WalkInOptionResponse {
    private List<OptionGroup> groups;

    @Data
    @Builder
    public static class OptionGroup {
        // "Ưu tiên (Trống hoàn toàn)" hoặc "Dự phòng (Vướng lịch sau)"
        private String groupName;
        private List<TableOption> options;
    }

    @Data
    @Builder
    public static class TableOption {
        private List<Long> tableIds;
        private int totalCapacity;
        // FULL_AVAILABLE, MERGE_AVAILABLE, PARTIAL_AVAILABLE, PARTIAL_MERGED_AVAILABLE
        private String type;
        // null nếu là FULL_AVAILABLE hoặc MERGE_AVAILABLE
        private LocalDateTime availableUntil;
    }
}

File: desktop-staff/reservation-winforms/DTO/reservation/WalkInOptionResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace reservation_winforms.DTO.reservation
{
    public class WalkInOptionResponse
    {
        [JsonProperty("groups")]
        public List<OptionGroup> Groups { get; set; }

        public class OptionGroup
        {
            [JsonProperty("groupName")]
            public string GroupName { get; set; }

            [JsonProperty("options")]
            public List<TableOption> Options { get; set; }
        }

        public class TableOption
        {
            [JsonProperty("tableIds")]
            public List<long> TableIds { get; set; }

            [JsonProperty("totalCapacity")]
            public int TotalCapacity { get; set; }

            [JsonProperty("type")]
            public string Type { get; set; }

            [JsonProperty("availableUntil")]
            public DateTime? AvailableUntil { get; set; }
        }
    }
}